### ISSUE TITLE
SidePanel cannot be closed on first load without page flashing

### ISSUE THREAD

**Issue by obedparla** [heetch/flamingo#122]
## Issue
If I want to use the `SidePanel` component with a default state of `closed` (open = false), it will show itself for a moment and then hide, which causes an ugly flash in the first render.

This can be fixed by making everything inside of it `display: none` if `open = false`. 

I fixed this in the admin by creating an extra component around the SidePanel and adding:

`  // hide all the panel elements on first load. Flamingo doesn't do this
  .f-Overlay,
  .f-SidePanel {
    display: ${({ isOpen }) => (isOpen ? 'block' : 'none')};
  }
`

This could be nicely fixed when moving the Sidepanel to Styled Components.

### Usage with react-router

I'm using the sidepanel with React Router now, and since the router will mount-dismount the component, I can't rely on "isOpen" prop anymore to use the animation. It would be awesome to add the functionality for the animation to trigger on `mount/dismount` rather than by props. 

I could migrate the component to SC and take care of this ^

**Comment by obedparla** [comment]
An even better solution would be to _not_ render anything inside of the SidePanel until `open === true`. This is easy to do, but in order to add the opening animation something like `react-transition-group`

**Comment by ugogo** [comment]
👋 @obedparla 
The `SidePanel` has an `isOpen` prop set to `false` by default and we never experienced any flash 🤔 
Could you provide a screencast?
Thanks!

**Comment by obedparla** [comment]
I'm sending `isOpen` as `false`, but I still get a flashing animation of it hiding. See here https://share.getcloudapp.com/kpuYepJr. 

I had to add the CSS above in order to completely hide it. That was the only way I could prevent the flashing. 

The CSS in the component is missing a "hide" if `open` is false, or something similar. So it shows and then does the animation of hiding. At least that's what I figured. Here's the code I have to run the video above: 
![image](https://user-images.githubusercontent.com/10674462/77427087-0a369680-6dd6-11ea-8691-d19a6b16196c.png)
